fhttp/http_client.go

```diff
@@ -709,13 +719,24 @@ func (c *FastClient) Fetch() (int, []byte, int) {
 	c.headerLen = 0
 	// Connect or reuse existing socket:
 	conn := c.socket
-	reuse := (conn != nil)
+
+	reuse := conn != nil
+	if c.maxConnectionReuse != 0 && c.currentConnection >= c.maxConnectionReuse {
+		reuse = false
+		c.currentConnection = 0
+		log.LogVf("[%d] Thread reach the threshold for max connection reuse of %d, force create new connection",
+			c.id, c.maxConnectionReuse)
+	}
+
 	if !reuse {
 		conn = c.connect()
```

Comment:
I'd maybe put the  `c.reuseCount=0` here and not worry about only incrementing it if the feature is on; it's a variable on the thread and just an increment and could be useful info to see

Reply:
fixed